aws/resource_aws_cloudwatch_event_bus_policy_test.go

```diff
@@ -0,0 +1,98 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	events "github.com/aws/aws-sdk-go/service/cloudwatchevents"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccAWSCloudwatchEventBusPolicy_basic(t *testing.T) {
+	resourceName := "aws_cloudwatch_event_bus_policy.test"
+	rstring := acctest.RandString(5)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSCloudWatchEventBusDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSCloudwatchEventBusPolicyConfig(rstring),
+				Check: resource.ComposeTestCheckFunc(
```

Comment:
Could you check for properties in the policy to confirm they match key details like principal and statement sid?

e.g. `resource.TestCheckResourceAttr(resourceName1, "principal", principal),`

Reply:
The problem with this check is that we are setting the whole policy JSON text as a field in the "event_bus_policy" resource configuration. We are therefore not able to check fields individually, we can just compare the JSON document in the state with a reference. Since this JSON is generated at runtime we are not able to set such reference as a parameter for the test function.
We found a similar scenario for the "[aws_iam_role](https://github.com/Cazoo-uk/terraform-provider-aws/blob/42c30754145d2a5a1bfb2a3eb5e765a5e9caa782/aws/data_source_aws_iam_role_test.go#L70)". In this case the tests make use of a data source for re-importing the configuration of the generated resource into the state and compare its fields with the ones of the previously generated resource. Implementing a solution like this would require us to implement the "event_bus_policy" data source.
Is there a different way to handle this case?

Reply:
https://github.com/hashicorp/terraform-provider-aws/pull/16874/files?file-filters%5B%5D=.go#diff-23812e233d355579267c55e833a81e4aa3a286298ee1477b978cbcd22ec7ad0cR141

Checks all of it, so we're good here.